src/EntityFramework/Migrations/Infrastructure/EdmModelDiffer.cs

```diff
@@ -328,6 +328,27 @@ where m1.EdmEquals(m2)
                     / (entityType1.DeclaredMembers.Count + entityType2.DeclaredMembers.Count)) > 0.80;
         }
 
+        private bool SourceAndTargetMatch(EntityType sourceEntityType, EntityTypeMapping sourceEntityTypeMapping, EntityType targetEntityType, EntityTypeMapping targetEntityTypeMapping)
+        {
+            return (sourceEntityTypeMapping.EntityType != null
+                    && targetEntityTypeMapping.EntityType != null
+                    && sourceEntityType == sourceEntityTypeMapping.EntityType
+                    && targetEntityType == targetEntityTypeMapping.EntityType)
+                   || (sourceEntityTypeMapping.EntityType == null
+                       && targetEntityTypeMapping.EntityType == null
+                       && sourceEntityTypeMapping.IsOfTypes.Contains(sourceEntityType)
```

Comment:
This bit of code I did not want to redo, we tried multiple versions where it would break other tests. I opted to leave it alone, albeit sticking it in a function rather than attempt to unroll it.

Honestly this entire decision should be a separate class with some descriptive tests to understand the different situations it should take into account. Given that does not exist I opted to leave that code alone.

I'd be happy to improve it with some requirements, but with the hours invested in playing with it I opted to leave it alone.